### ISSUE TITLE
No way to run command, and get both it's output and it's return code

### ISSUE THREAD

**Issue by shoham-stratoscale** [openresty/lua-nginx-module#779]
I need, in my lua script, to run a shell command. Then, I need to check it succeeded, using it's retval, and get it's stdout.
To achieve that, I have this code:
```lua
handler = assert(io.popen("echo my command is here"))
data = handler:read("*a")
succeeded, error_msg, retcode = handler:close()
```

"data" indeed gets the command output, but "succeeded" is always nil.
"error_msg" is "No child processes", and "retcode" is 10

**Comment by jsopenrb** [comment]
That happends because nginx has a handler for `SIGCHLD`. As far as I know the only way to fix this is to patch nginx core.

**Comment by agentzh** [comment]
See https://github.com/juce/lua-resty-shell for a solution for this.

**Comment by shoham-stratoscale** [comment]
@agentzh Thank you for your response.
I'm familiar with the lua-resty-shell, and it's a great solution for not-blocking the nginx worker.
But it doesn't solve this bug - I just want a simple way to run a command and get it's return code. I don't think starting another process/server (sockproc) and talking to it is simple or fit in most use-cases.
It's much better if io:popen would simply work as expected.

**Comment by agentzh** [comment]
@shoham-stratoscale Yes, we could "fix" up nginx's SIGCHLD signal handler in ngx_lua ourselves. Patches welcome.
